### PR TITLE
[function](vec)use const_col to opt current functions.

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed Changes

before:

```
mysql [ssb]>select count(now()) from lineorder;
+--------------+
| count(now()) |
+--------------+
|    600037902 |
+--------------+
1 row in set (0.96 sec)
```

```
mysql [ssb]>select count(curdate()) from lineorder;
+------------------+
| count(curdate()) |
+------------------+
|        600037902 |
+------------------+
1 row in set (0.95 sec)
```

after:
```
mysql [ssb]>select count(now()) from lineorder;
+--------------+
| count(now()) |
+--------------+
|    600037902 |
+--------------+
1 row in set (0.85 sec)
```
```
mysql [ssb]>select count(curdate()) from lineorder;
+------------------+
| count(curdate()) |
+------------------+
|        600037902 |
+------------------+
1 row in set (0.88 sec)
```

## Problem Summary

The changes are like https://github.com/apache/doris/pull/16953 , apply to CurrentDate and CurrentDateTime.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run p0

**Comment by cambyzju** [comment]
Could you also give some results before and after, like pr https://github.com/apache/doris/pull/16953

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run p0

File: be/src/vec/functions/function_date_or_datetime_computation.h [hunk 1]
```diff
@@ -847,6 +846,7 @@ struct CurrentDateTimeImpl {
                               size_t input_rows_count) {
         auto col_to = ColumnVector<NativeType>::create();
         DateValueType dtv;
+        bool _const = false;
```
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:849]: do not use `_name` in not class member.

Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:849]: do not use `_name` in not class member.

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**